### PR TITLE
Lubby2 rate form

### PR DESCRIPTION
**Pull request by zhangning737**
@nagelt changed Lubby2 to rate form, tested by sample shear benchmark.

### CODE REVIEW

**Comment by wenqing** [comment]
Benchmark failed.

Do you need to change the documentation (https://www.opengeosys.org/docs/benchmarks/small-deformations/mechanics-lubby2-shear-traction/) as well?

**Comment by zhangning737** [comment]
@wenqing yes, the documentation should also be changed in several equations...

**Comment by zhangning737** [comment]
@wenqing the documentation has been updated...

**Comment by zhangning737** [comment]
@wenqing added Jacobian expression at the last push, pls have a check...

**Comment by wenqing** [comment]
Documentation: In the sentence above eqn.(4), epsilon_M^{DT} and epsilon_K^{DT} has to be swapped. See.
Untitled

It can be merged once this typo is corrected.

**Comment by zhangning737** [comment]
@wenqing exactly as you said, changed...

File: MaterialLib/SolidModels/Lubby2.cpp [hunk 1]
```diff
@@ -94,9 +94,11 @@ Lubby2<DisplacementDim>::integrateStress(
     // calculation of deviatoric parts
     auto const& P_dev = Invariants::deviatoric_projection;
     KelvinVector const epsd_i = P_dev * eps;
+    KelvinVector const epsd_t = P_dev * eps_prev;
 
-    // initial guess as elastic predictor
+    // initial guess as elastic predictor. Note: dimensionless stresses!
```
Comment (wenqing) [inline MaterialLib/SolidModels/Lubby2.cpp]: Is this statement "Note: dimensionless stresses!"  correct? Other wise please drop it.

File: MaterialLib/SolidModels/Lubby2.cpp [hunk 2]
```diff
@@ -210,7 +216,7 @@ void Lubby2<DisplacementDim>::calculateResidualBurgers(
 {
     // calculate stress residual
     res.template segment<KelvinVectorSize>(0).noalias() =
-        stress_curr - 2. * (strain_curr - strain_Kel_curr - strain_Max_curr);
+        (stress_curr - stress_t)/dt - 2./dt * ((strain_curr - strain_t) - (strain_Kel_curr - strain_Kel_t) - (strain_Max_curr - strain_Max_t));
```
Comment (wenqing) [inline MaterialLib/SolidModels/Lubby2.cpp]: If you can run clang.format, please apply it here. Otherwise, just leave it as it.

**Comment by wenqing** [comment]
:fast_forward:

File: MaterialLib/SolidModels/Lubby2.cpp (hunk 2 above)
Comment (zhangning737) [reply MaterialLib/SolidModels/Lubby2.cpp]: My clang.format in the vscode editor seems making the code more ugly, so please pull it first, need time to fix the editor's problem...

Comment (wenqing) [reply MaterialLib/SolidModels/Lubby2.cpp]: Just sent a PR about clang-format to your branch, please merge it.

Comment (zhangning737) [reply MaterialLib/SolidModels/Lubby2.cpp]: nice, already merged...

**Pull request merged**